src/reducers/dependencies.reducer.js

```diff
@@ -120,6 +126,91 @@ export default (state: State = initialState, action: Action) => {
       });
     }
 
+    case ADD_DEPENDENCIES_START: {
+      const { projectId, dependencies } = action;
+
+      return produce(state, draftState => {
+        // this action name is a bit of a misnomer, since this block
+        // is used for both adding AND updating dependencies (otherwise,
+        // we wouldn't be able to batch installs/updates with a single
+        // `npm install` command
+
+        dependencies.forEach(dependency => {
+          draftState[projectId][dependency.dependencyName] = {
+            name: dependency.dependencyName,
+            status: dependency.updating ? 'updating' : 'installing',
+            // If we're installing a new dependency, set all other
+            // fields to empty strings to prevent issues with nullable
+            // values. Otherwise we're updating, so don't overwrite
+            // anything else.
+            ...(dependency.updating
+              ? {}
+              : {
+                  description: '',
+                  version: '',
+                  homepage: '',
+                  license: '',
+                  repository: '',
+                }),
```

Comment:
I think to avoid this, we could do:

```
name: dependency.dependencyName,
status: 'installing',
description: dependency.description || '',
version: dependency. version || '',
// and so on
```

Reply:
The current iteration of this reducer isn't passed a full version of the dependency, but instead a stripped-down version that only includes `dependencyName` and optionally `version`. I could probably rewrite it to include that other info, but I figured that other than in this block that would be extraneous and it would be better to have this one instance of wonky syntax than pass around all that extra information needlessly.

Reply:
The current iteration of this reducer isn't passed a full version of the dependency, but instead a stripped-down version that only includes `dependencyName` and optionally `version`. I could probably rewrite it to include that other info, but I figured that other than in this block that would be extraneous and it would be better to have this one instance of wonky syntax than pass around all that extra information needlessly.